src/pixi/IgePixiAnimation.js

```diff
@@ -88,25 +88,29 @@ var IgePixiAnimation = IgeClass.extend({
 	},
 	select: function (frames, fps = 15, loopCount, cellSheetAnimId, animName) {
 		var self = this;
-		var entity = ige.pixi.trackEntityById[self._entity.entityId];
-		if (!entity) return;
+		// this was some patchy code. I propose a solution that we can work with.
+		if (
+			ige.entitiesToRender.trackEntityById[self._entity.entityId] &&
+			ige.entitiesToRender.trackEntityById[self._entity.entityId]._pixiContainer
+		) {
 
-		var startFrame = frames[0] - 1;
-		var lastFrame = frames[frames.length - 1] - 1;
+			var startFrame = frames[0] - 1;
+			var lastFrame = frames[frames.length - 1] - 1;
 
-		self.i = 0;
-		self.fpsSecond = 1000 / fps;
-		self.frames = frames;
-		self.loopCount = loopCount;
-		self.lastFrame = lastFrame;
-		self.startFrame = startFrame;
-		self.totalNumberOfFrames = frames.length;
-		self._entity.currentAnimId = cellSheetAnimId;
+			self.i = 0;
+			self.fpsSecond = 1000 / fps;
+			self.frames = frames;
+			self.loopCount = loopCount;
+			self.lastFrame = lastFrame;
+			self.startFrame = startFrame;
+			self.totalNumberOfFrames = frames.length;
+			self._entity.currentAnimId = cellSheetAnimId;
 
-		this.resetAnimation();
+			self.resetAnimation();
```

Comment:
don't see the need to use `self` in this method at all, there are no callbacks that have to preserve context

Reply:
sounds good. I swapped the last few to `self` because it was a mix